src/coin_test/backtest/portfolio.py

```diff
@@ -44,3 +47,31 @@ def available_assets(self, asset: Ticker) -> Money:
             raise ValueError("This asset does not exist in the portfolio.")
 
         return self.assets[asset] - self.reserved[asset]
+
+    def adjust(self, trade: Trade) -> Union["Portfolio", None]:
+        """Adjust the portfolio after a given Trade is performed.
+
+        Args:
+            trade: The Trade object that is completed
+
+        Returns:
+            Portfolio | None: The new adjusted portfolio or None if insufficient Money
+        """
+        adjusted_assets = self.assets.copy()
+
+        currency = trade.asset_pair.currency
+        asset = trade.asset_pair.asset
+        if trade.side == Side.BUY:
+            # Check for overspending
+            if adjusted_assets[currency] < Money(currency, trade.amount * trade.price):
+                return None
+
+            adjusted_assets[currency] -= Money(currency, trade.amount * trade.price)
+            adjusted_assets[asset] += Money(asset, trade.amount)
+        else:
+            if adjusted_assets[asset] < Money(asset, trade.amount):
```

Comment:
See other comment - if the trade has already been confirmed to run, there shouldn't need to be any validation here